### DISCUSSION TITLE
iCub automatic driving within the Carla simulation environment

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/community#692] [Show and tell]
Hi, 

Recently, I have been working on implementing iCub automatic driving within the Carla simulation environment. The steering commands are computed using a PID controller, and communication is facilitated through ROS messages. The wheel grasping and manipulation techniques have been adopted from the icub-gazebo-grasping-sandbox project.

Below is what i have done.

https://github.com/robotology/community/assets/7019692/26b17055-0412-4f82-9245-d3ac56f7f034

**Comment by pattacini** [comment]
Interesting evolution of autonomous driving cars 😉
